src/Substructure.jl

```diff
@@ -0,0 +1,313 @@
+- `clusterseq`: The cluster sequence from which the original jet is obtained.
+- `rad`: The new jet radius.
+- `mtd`: The new clustering method.
+
+# Returns
+- ClusterSequence: The new cluster sequence.
+"""
+function recluster(jet::PseudoJet, clusterseq::ClusterSequence, rad = 1.0, mtd = 0)
+    cons = constituents(jet, clusterseq)
+    new_clusterseq = jet_reconstruct(cons; p = mtd, R = rad, strategy = RecoStrategy.N2Plain)
+   
+    return new_clusterseq
+end
+
+function sort_jets!(event_jet_array)
+    jet_pt(jet) = pt2(jet)
+    sort!(event_jet_array, by = jet_pt, rev = true)
+    # println(event_jet_array)
+end
+
+function join(jets::Vector{PseudoJet})
+    final = PseudoJet(0.0, 0.0, 0.0, 0.0)
+    
+    for jet in jets
+        final = final + jet
+    end
+
+    final
+end
+
+# Defining suitable structures to be used for jet grooming and tagging
+"""
+    struct MassDropTagger
+
+Used for tagging jets that undergo mass drop, a common technique in jet substructure.
+
+Fields:
+- mu: Maximum allowed mass ratio for a jet to pass tagging
+- y: Minimum kT distance threshold for parent separation
+"""
+
+mutable struct MassDropTagger
+    mu::Float64
+    y::Float64
+end
+
+"""
+    struct SoftDropTagger
+
+Applies a soft-drop condition on jets, trimming away soft, wide-angle radiation.
+
+Fields:
+- zcut: Minimum allowed energy fraction for subjets
+- b: Angular exponent controlling soft radiation suppression
+"""
+
+mutable struct SoftDropTagger
+    zcut::Float64
+    b::Float64
+end
+
+"""
+    struct Filter
+
+Filters jets based on radius and number of hardest subjets, reducing contamination.
+
+Fields:
+- filterRadius: Radius parameter to recluster subjets
+- numHardestJets: Number of hardest jets to retain in the filtered result
+"""
+
+mutable struct Filter
+    filterRadius::Float64 
+    numHardestJets::Int
+end
+
+"""
+    struct Trim
+
+Trims soft, large-angle components from jets based on fraction and radius.
+
+Fields:
+- trimRadius: Radius used for reclustering in trimming
+- trimFraction: Minimum momentum fraction for retained subjets
+- reclusterMethod: Method identifier for reclustering
+"""
+
+mutable struct Trim
+    trimRadius::Float64
+    trimFraction::Float64
+    reclusterMethod::Int
+end
+
+"""
+    massDrop(jet, clusterseq, tag) -> PseudoJet
+
+Identifies subjets in a jet that pass the mass drop tagging condition.
+The method stops at the first jet satisfying the mass and distance thresholds.
+
+Arguments:
+- jet: PseudoJet instance representing the jet to tag
+- clusterseq: ClusterSequence with jet clustering history
+- tag: MassDropTagger instance providing mass drop parameters
+
+Returns:
+- PseudoJet: The jet (or subjet) satisfying the mass drop conditions, if tagging is successful, otherwise a zero-momentum PseudoJet
+"""
+
+function massDrop(jet::PseudoJet, clusterseq::ClusterSequence, tag::MassDropTagger)
+    allJets = clusterseq.jets
+    hist = clusterseq.history
+
+    while(true)
```

Comment:
Don't use unnecessary parentheses: `while true` is good enough.

https://docs.julialang.org/en/v1/manual/style-guide/#Don't-parenthesize-conditions